infrastructures/docker/src/main/java/org/eclipse/che/workspace/infrastructure/docker/server/ServerChecker.java

```diff
@@ -0,0 +1,118 @@
+
+    private final String                    machineName;
+    private final String                    serverRef;
+    private final long                      period;
+    private final long                      deadLine;
+    private final CompletableFuture<String> reportFuture;
+    private final Timer                     timer;
+
+    /**
+     * Creates server checker instance.
+     *
+     * @param machineName
+     *         name of machine to whom the server belongs
+     * @param serverRef
+     *         reference of the server
+     * @param period
+     *         period between unsuccessful availability checks, measured in {@code timeUnit}
+     * @param timeout
+     *         max time allowed for the server availability checks to last before
+     *         server is treated unavailable, measured in {@code timeUnit}
+     * @param timeUnit
+     *         measurement unit for {@code period} and {@code timeout} parameters
+     */
+    protected ServerChecker(String machineName,
+                            String serverRef,
+                            long period,
+                            long timeout,
+                            TimeUnit timeUnit,
+                            Timer timer) {
+        this.machineName = machineName;
+        this.serverRef = serverRef;
+        this.timer = timer;
+        this.period = TimeUnit.MILLISECONDS.convert(period, timeUnit);
+        this.reportFuture = new CompletableFuture<>();
+        this.deadLine = System.currentTimeMillis() + TimeUnit.MILLISECONDS.convert(timeout, timeUnit);
+    }
+
+    /**
+     * Starts server availability checking, which will be stopped when server become available or
+     * checking times out.
+     */
+    public void start() {
+        timer.schedule(new ServerCheckingTask(), 0);
+    }
+
+    /**
+     * Shows whether the server is treated as available.
+     *
+     * @return true if server is available, false otherwise
+     */
+    public abstract boolean isAvailable();
+
+    /**
+     * Returns {@code CompletableFuture} that will be completed when server become available or unavailable.
+     * </p>
+     * When server become available completable future returns server reference.
+     * </br>This completable future can be used to chain {@code CompletableFuture} stage
+     * to successful or unsuccessful completion of the server availability check.
+     *
+     * @see CompletableFuture
+     * @see CompletableFuture#thenAccept(Consumer)
+     * @see CompletableFuture#exceptionally(Function)
+     */
+    public CompletableFuture<String> getReportCompFuture() {
+        return reportFuture;
+    }
+
+    private boolean isTimedOut() {
+        return System.currentTimeMillis() > deadLine;
+    }
+
+    private class ServerCheckingTask extends TimerTask {
+        @Override
+        public void run() {
+            LOG.info("Check server {} of machine {} at {}", serverRef, machineName, System.currentTimeMillis());
```

Comment:
Looks like this kind of information which can spam master logs, let's change the log level to _debug_?
More over consider removing ` at {}` as usually log frameworks add time automatically.